airflow/bin/cli.py

```diff
@@ -424,6 +424,50 @@ def clear(args):
         confirm_prompt=not args.no_confirm,
         include_subdags=not args.exclude_subdags)
 
+def mark_success(args):
+    dag = get_dag(args)
+    dagruns = dag.get_dagruns(args.start_date, args.end_date)
+    # todo: run.dag is transient, needs to be set
+    for run in dagruns:
+        run.dag = dag
+
+    tis = []
+    for run in dagruns:
+        tis += run.mark_success(
```

Comment:
I agree it's a little annoying, plus it creates an illusion of what is selected the first pass is definitely what is marked success (i.e. rare corner case which that isn't true is if 'future' is selected, and a TI that instantiate after the first pass will get marked success without being listed).